src/components/data-processing/DataIntegration/CalculationConfig.jsx

```diff
@@ -89,6 +91,10 @@ const CalculationConfig = (props) => {
     seuratv4: () => <SeuratV4Options config={dataIntegration.methodSettings.seuratv4} onUpdate={updateSettings} onChange={() => setChangesOutstanding(true)} />,
   };
 
+  const variationExplained = data.slice(0, dimensionalityReduction.numPCs).reduce((acum, current) => acum + current.percentVariance, 0);
+  const roundingPrecision = 6;
+  const roundedVariationExplained = _.round(variationExplained * 100, roundingPrecision);
```

Comment:
as I understand it, `variationExplained` and `roundingPrecision` variables are only used for calculating `roundedVariationExplained`. What I would do is to change `roundedVariationExplained` to a function that returns the rounded variation and then move `variationExplained` and `roundingPrecision` to be inside that function, since they are only needed there. Then in `InputNumber` for `value` just make a call to that function

Reply:
> I just checked the link in staging. It looks better, but I have a question: right now, if I change the `number of principal components` number, after clicking anywhere on the screen the red line updates even though the settings are not actually applied. Is that expected behaviour?
@ivababukova 

yes, [here](https://docs.google.com/presentation/d/1-6mtMrZF14GRAYtLjh3F7FmhyE4o5WRpwyxt5gISGDo/edit#slide=id.gb76744ee2d_0_247) it says that "When the user changes this value, the red line in the Elbow Plot (left) will update"